src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final RegistrationService registrationService = new RegistrationServiceImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_LENGTH = 5;
+    private static final int MIN_PASSWORD_LENGTH = 5;
+    private static final String LOGIN_WITH_MIN_LENGTH = "io0p1";
+    private static final String PASSWORD_WITH_MIN_LENGTH = "12345";
+    private final StorageDao storageDao = new StorageDaoImpl();
+
+    @Test
+    public void registerNullNotOk() {
+        assertThrows(RegistrationException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    public void registerNullLoginNotOk() {
+        User user = getValidUser();
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void registerNullPasswordNotOk() {
+        User user = getValidUser();
+        user.setPassword(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void registerNullAgeNotOk() {
+        User user = getValidUser();
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    public void registerAddLoginNotOk() {
+        User user = getValidUser();
+        StringBuilder login = new StringBuilder();
+        user.setLogin(login.toString());
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        for (int i = 1; i < MIN_LOGIN_LENGTH; i++) {
+            user.setLogin(login.append(i).toString());
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    public void registerAlredyRegisteredUserNotOk() {
+        User user = getValidUser();
+        storageDao.add(user);
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+
+    }
+
+    @Test
+    public void registerAddPasswordNotOk() {
+        User user = getValidUser();
+        StringBuilder password = new StringBuilder();
+        user.setPassword(password.toString());
+
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        for (int i = 1; i < MIN_PASSWORD_LENGTH; i++) {
+            user.setPassword(password.append(i).toString());
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    public void registerAddAgeNotOk() {
+        User user = getValidUser();
+
+        for (int i = 0; i < MIN_AGE; i++) {
+            user.setAge(i);
+            assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        }
+    }
```

Comment:
The test `registerAddAgeNotOk` should be named following the convention, such as `register_ageBelowMin_notOk`. Also, this test should include the edge case of `MIN_AGE - 1` to ensure that the boundary condition is tested properly.